### PR TITLE
Implementation of Video Search Use Case (does not include Tests)

### PR DESCRIPTION
**Pull request by caratooo**
This merge includes the full implementation of the Video Search use case, without the tests.

The use case allows users to input any Youtube video ID and get the statistics of the video. Some example statistics are: view count, like count, uploading channel name, etc.

**Note**: 
- In addition to the basic functionalities of this use case, this merge also includes changes to the History use case to ensure to connection between History and Video Search.
- Comprehensive tests will be included in another merge.
- GUI may need some tweaking...

### CODE REVIEW

File: src/use_case/video_search/VideoSearchInteractor.java
Comment (jonathanjschen913) [inline src/use_case/video_search/VideoSearchInteractor.java:1]: Remember to clean up your comments for unused imports!

**Comment by cyn900** [comment]
This use case is working:
- a new CSV file is added when the user signs up
- history shows the correct thing

**Things to improve**:
- need more documentation
- the history view is not showing entirely

File: src/use_case/video_search/VideoSearchInteractor.java
Comment (caratooo) [reply src/use_case/video_search/VideoSearchInteractor.java:1]: just did!

File: src/views/VideoStatsView.java
Comment (jonathanjschen913) [inline src/views/VideoStatsView.java:1]: Remember to clean up the comments! The panels can also use some more meaningful naming, but functionally speaking it is working.

File: test/views/TrendingCategorySelectViewTest.java
Comment (jonathanjschen913) [inline test/views/TrendingCategorySelectViewTest.java:1]: Remember to clean up comments

File: src/use_case/video_search/VideoSearchInteractor.java
Comment (AzkyT) [inline src/use_case/video_search/VideoSearchInteractor.java:1]: There's a TODO comment here on line 45

**Pull request merged**